infra/modules/identity-provider/resources/main.tf

```diff
@@ -4,6 +4,14 @@
 ## - Configures MFA
 ############################################################################################
 
+locals {
+  dash_domain = var.domain_name != null ? replace(var.domain_name, ".", "-") : null
+}
+
+data "aws_sesv2_email_identity" "main" {
+  email_identity = var.domain_name
+}
```

Comment:
i regret asking you to remove the extra variable in exchange for the data source, can we revert this particular change